### PR TITLE
 ci(dep): Add step to commit changes if PR has dependencies label

### PR DESCRIPTION
**Pull request by sayboras**
This commit is to perform below steps:
    
- If PR is having dependencies label, update files in dist as well
- If PR doesn't have dependencies label, fail the build if there
is dirty changes.

Assumption:
- dependabot PR is always having label dependencies 
- I get username and email from commit log, hope that it will not change. No token is required.
- Verification will be done after this PR is merged only.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -14,7 +14,29 @@ jobs:
...
+      - name: Update dist files
+        if: github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'dependencies')
+        run: |
+          if [[ -z $(git status -s) ]]
+          then
+            echo "No change is required"
+          else
+            echo "Updating dist directory"
+            git config --local user.name "dependabot[bot]"
+            git config --local user.email "49699333+dependabot[bot]@users.noreply.github.com"
```
Comment (SVilgelm) [inline .github/workflows/test.yml:28]: I think it's better to use the `goreleaserbot <goreleaser@carlosbecker.com>`
```suggestion
            git config --local user.name "goreleaserbot"
            git config --local user.email "goreleaser@carlosbecker.com"
```

**Review by SVilgelm — CHANGES_REQUESTED** [review]

**Comment by SVilgelm** [comment]
lets merge it after v2.4.0 released and then we can test it, I don't see any problems here, if it's not working, then we can revert

**Comment by butuzov** [comment]
@SVilgelm is time to merge (we are currently past due `v2.4.0`)?

**Comment by SVilgelm** [comment]
finally, let's merge it, more than a year when changes are done :)

**Review by SVilgelm — APPROVED** [review]

**Pull request merged**